compiler/luci/service/src/CircleShapeInferenceRule.cpp

```diff
@@ -2019,6 +2025,30 @@ class ShapeInferenceAlgorithm final : public luci::CircleNodeVisitor<loco::NodeS
     return loco::NodeShape{*then_graph_output->shape()};
   }
 
+  loco::NodeShape visit(const luci::CircleNonMaxSuppressionV4Out *node) final
+  {
+    const loco::DataType S32 = loco::DataType::S32;
+
+    auto nmsv4 = dynamic_cast<const luci::CircleNonMaxSuppressionV4 *>(node->input());
+    if (nmsv4 == nullptr)
+      INTERNAL_EXN("CircleNonMaxSuppressionV4 IR is not configured correctly");
+
+    auto index = node->index();
+    if (index == 1)
+      return loco::TensorShape({0});
+
+    assert(index == 0);
+
+    auto unknown = loco::TensorShape{loco::Dimension()};
+    auto max_output_size = dynamic_cast<const luci::CircleConst *>(nmsv4->max_output_size());
+    if (max_output_size == nullptr || max_output_size->size<S32>() < 1)
+      return unknown; // we need CircleConst for max output size
+    LUCI_ASSERT(max_output_size->dtype() == S32, "Only support int32 for max_output_size");
```

Comment:
This assert is duplicated in line 2044 (https://github.com/Samsung/ONE/pull/3652/files#diff-aada5b5648e693d6ff6ca4b253f1b255R2044) in `size<S32>` call.
If you want to check type, please move this assert before the condition.